python/tchannel/tornado/tchannel.py

```diff
@@ -103,15 +108,24 @@ def __init__(self, name, hostport=None, process_name=None,
         )
 
         self.name = name
+        self._trace = trace
 
         # register event hooks
         self.event_emitter = EventEmitter()
         self.hooks = EventRegistrar(self.event_emitter)
+        self.hooks.register(ZipkinTraceHook(tchannel=self))
```

Comment:
Hm. I don't like the idea of auto-registering hooks like this. Would this even work if I wanted to use tchannel without a zipkin collector setup?